components/sync15/ios/RustSyncTelemetryPing.swift

```diff
@@ -0,0 +1,622 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
```

Comment:
sheesh, it really sucks that we still need this hackery (but yeah, I agree we do - it just sucks :)

Reply:
Yeah, sadly--particularly if we want to create more/better graphs with this telemetry data. I also think I might need to add some unit tests here (although there aren't any for the android logic this file was based on).